lib/actions/field-trip.js

```diff
@@ -520,6 +523,16 @@ function prepareQueryParams(request, outbound) {
         OTP_API_TIME_FORMAT
       )
     }
+    // Generate banned list from clear list
+    if (fieldTripOptions?.allowedAgencies) {
+      const bannedAgencies = transitOperators
+        .map((to) => to.agencyId)
+        .filter((to) => !fieldTripOptions.allowedAgencies.includes(to))
```

Comment:
If you `filter` first, the second loop will have to loop over fewer items.

Reply:
You are right, but cleaner code is worth the tradeoff for this tiny list